### PR TITLE
Vector3i usability improvements

### PR DESCRIPTION
**Pull request by Josharias**
Rename org.terasology.math.Vector3i into org.terasology.math.Vector3iUtil and make all its methods static. This leaves all Vector3i code to use only the TeraMath Vector3i and eliminate the casting problems when using results from methods.

The main problem with the current org.terasology.math.Vector3i is that after creating one and you call a method like .add(..) and it returns you a org.terasology.math.geom.Vector3i which no longer is useable if the surrounding code is expecting the original type.

In addition to Vector3iUtil only containing static methods, the old Vector3i.toVector3f() method had to be replaced as well. The toVector3f() functionality is found in Vector3fUtil.

I have a large number of the modules in my working copy as well and will make appropriate PRs on those repos as well in the next day or so.

### CODE REVIEW

**Comment by msteiger** [comment]
Great stuff! I think that several methods are still missing in the TeraMath version of Vector3i. There's an open PR that aims to add them:

https://github.com/MovingBlocks/TeraMath/pull/26

How did you tackle the problem of missing methods? Maybe we should merge that PR first, so we can benefit from the additional methods (min, max, div, one, zero, east, west, north, south ... ). 

I can help with the modules - maybe we can coordinate that real quick on Slack/IRC?

**Comment by skaldarnar** [comment]
I encountered similar issues and directly used the TeraMath implementation for my latest PRs. We need to make sure to adapt all methods taking a Vector3i as argument as well (e.g., encompasses() of Region3i, ...). 

I'd like to follow (and participate) in the upcoming discussion on Slack/IRC ;)

**Comment by Josharias** [comment]
Looks like the PR from TeraMath should indeed solve some of these Vector3i shortcomings.  Looks like you have even dealt with the .toVector3f().  Sweet.

I'll see if I can get TeraMath on my workstation today so it can continue.  

I think the modules will be pretty straightforward.  The refactoring tools in intellij have taken care of most of it already.  I just have to do some git work to get branches created and pushed.

**Comment by Josharias** [comment]
Missing methods I dealt with by turning all the methods static and adding a first parameter that is used instead of "this" keyword.

File: modules/Core/src/main/java/org/terasology/core/world/generator/facetProviders/SurfaceObjectProvider.java [hunk 1]
```diff
@@ -54,10 +55,11 @@ public void setSeed(long seed) {
 
     /**
      * Populates a given facet based on filters and population densities
-     * @param facet the facet to populate
+     *
+     * @param facet        the facet to populate
      * @param surfaceFacet the surface height facet
-     * @param typeFacet the facet that provides the environment
-     * @param filters a set of filters
+     * @param typeFacet    the facet that provides the environment
```
Comment (skaldarnar) [inline modules/Core/src/main/java/org/terasology/core/world/generator/facetProviders/SurfaceObjectProvider.java]: are these stylistic changes due to your setup? do we have a convention about indenting parameter descriptions?

Comment (Cervator) [inline modules/Core/src/main/java/org/terasology/core/world/generator/facetProviders/SurfaceObjectProvider.java]: IIRC I don't think we do and indentation has probably been inconsistently applied over time. I could go with either flavor, no preference

**Comment by Cervator** [comment]
I'll let you all worry about this one, I don't even dare open the files changed page lest my tab-heavy browser is sure to crash :D

I can check it out and see what modules crash sometime since I have so many locally. Let me know if I should do that sometime or if you all are on top of it :-)

File: modules/Core/src/main/java/org/terasology/core/world/generator/facetProviders/SurfaceObjectProvider.java (hunk 1 above)
Comment (Josharias) [inline modules/Core/src/main/java/org/terasology/core/world/generator/facetProviders/SurfaceObjectProvider.java]: Yeah,  this was all Intellij's doing.  I have a default style configuration save for expanding all imports.

**Comment by Josharias** [comment]
Will reopen with new TeraMath release.

**Pull request closed without merging**